client/src/lib.rs

```diff
@@ -406,7 +408,63 @@ impl<'a> RequestBuilder<'a> {
         self.signers.push(signer);
         self
     }
+    pub fn send_with_config(
+        self,
+        config: RpcSendTransactionConfig,
+    ) -> Result<Signature, ClientError> {
+        let accounts = match self.namespace {
+            RequestNamespace::State { new } => {
+                let mut accounts = match new {
+                    false => vec![AccountMeta::new(
+                        anchor_lang::__private::state::address(&self.program_id),
+                        false,
+                    )],
+                    true => vec![
+                        AccountMeta::new_readonly(self.payer.pubkey(), true),
+                        AccountMeta::new(
+                            anchor_lang::__private::state::address(&self.program_id),
+                            false,
+                        ),
+                        AccountMeta::new_readonly(
+                            Pubkey::find_program_address(&[], &self.program_id).0,
+                            false,
+                        ),
+                        AccountMeta::new_readonly(system_program::ID, false),
+                        AccountMeta::new_readonly(self.program_id, false),
+                    ],
+                };
+                accounts.extend_from_slice(&self.accounts);
+                accounts
+            }
+            _ => self.accounts,
+        };
+        let mut instructions = self.instructions;
+        if let Some(ix_data) = self.instruction_data {
+            instructions.push(Instruction {
+                program_id: self.program_id,
+                data: ix_data,
+                accounts,
+            });
+        }
+
+        let mut signers = self.signers;
+        signers.push(&self.payer);
+
+        let rpc_client = RpcClient::new_with_commitment(self.cluster, self.options);
 
+        let tx = {
+            let (recent_hash, _fee_calc) = rpc_client.get_recent_blockhash()?;
+            Transaction::new_signed_with_payer(
+                &instructions,
+                Some(&self.payer.pubkey()),
+                &signers,
+                recent_hash,
+            )
+        };
+        rpc_client
+            .send_transaction_with_config(&tx, config)
```

Comment:
This will not confirm the transaction, as the `send` counterpart below does. This will create an asymmetry between the two methods. We should probably confirm here as well.

Reply:
Unfortunately there is no `send_and_confirm_transaction_with_config` at least none that i could find in the docs

Reply:
given that this is a builder, wouldnt it make more sense to add an Option\<RpcSendTransactionsConfig> as a field and then only have 1 send method that matches on it? Like this:
```
match self.send_options {
   Some(config) => send_and_confirm_transaction_with_spinner_and_config(..),
   None => send_and_confirm_transaction(..),
}
```

`send_and_confirm_transaction_with_spinner_and_config` exists but adds a spinner while `send_and_confirm_transaction` does not but we could adjust that case to use `send_and_confirm_transaction_with_spinner`

```
match self.send_options {
   Some(config) => send_and_confirm_transaction_with_spinner_and_config(..),
   None => send_and_confirm_transaction_with_spinner(..),
}
```